src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi

```diff
@@ -165,3 +167,22 @@ async def generator_to_async_generator(object gen, object loop, object thread_po
 
     # Port the exception if there is any
     await future
+
+
+if PY_MAJOR_VERSION >=3 and PY_MINOR_VERSION >=7:
```

Comment:
I believe this is a compile-time option. Can you use `IF` here?

Reply:
IIRC this is what @gnossen is saying, of  so could we get this information using the `sys` module and under the context of the `get_working_loop`?

So having something like this

```python
def get_working_loop():
    import sys
    if sys.version >= (3, 7):
        return asyncio.get_running_loop()
    ...
```

Reply:
Not quite. Since this is going to be compiled into Cpython-version-specific C extension, [we can just compile in the correct definition](https://stackoverflow.com/questions/47276648/cython-precompiler-decision-making/47292793#47292793) instead of looking it up at runtime.

Reply:
The comment in source code recommended using `if/else` then `IF/ELSE` (https://github.com/cython/cython/blob/master/Cython/Includes/cpython/version.pxd#L3).

The C++ compiler is going to optimize this part anyway. The only difference is whether this check gonna appear in the generated cpp file.

Reply:
You are right @gnossen, thanks for the clarification.